### DISCUSSION TITLE
Feature request: Taildrive

### DISCUSSION THREAD

**Discussion by bjeanes** [hassio-addons/app-tailscale#391] [Ideas]
Adding support for [Taildrive](https://tailscale.com/kb/1369/taildrive) and, in particular, sharing the `/config` directory as a share would be great.

I have Tailscale on the Host OS too so I am already doing this myself, but I think it would be nice to have it baked into the add-on.

**Comment by lmagyar** [comment]
Hmmm, as I see, it has the same issue as the `tailscale serve` command: who wants to write the config UI for it, if Tailscale says they will do it "sometime in the future"? In case of `serve` and `funnel` the add-on has a bool switch and enables accessing HA's UI through serve/funnel. If somebody needs some extra TS (serve) config, can use the cli of TS inside the add-on and configure it freely.

It is not mentioned in the docs, so general users won't copypaste it and ruin eg., but you can access the add-on's cli from eg. the SSH add-on with ``docker exec -it `docker ps -q -f name=tailscale` /bin/bash`` and then you can configure `/opt/tailscale drive ...`. I haven't tested it, but assume it has "memory" ie. the drive config is saved in the TS state and will be used after an add-on restart, and even after an add-on backup/restore. It woth a try.

**Reply by bjeanes** [reply]
Yeah all good points. I think having a specific option to enable it and to specifically share `/config` would be sufficient. Like you gesture, people could then go and add extra shares manually if necessary. Sharing the config is, IMO, the main use-case.

I've been managing my config via Syncthing add-on for yonks but in the last few months I've been having it stop syncing silently (the UI shows that it sees the change, but the files are not modified). I've re-set up the Syncthing share like 5 times and then after a few days it stops.

So, I've switched to editing off `webdav://100.100.100.100/homeassistant-os/ha-config` but this relies on the OS Tailscale I have running. Having it built-in to the add-on would be ace.

**Reply by bjeanes** [reply]
Though, `/config` is not (yet?) mounted into this add-on anyway:

```sh-session
➜  ~ docker exec -it `docker ps -q -f name=tailscale` mount | grep -E 'homeassistant|config'
➜  ~ # ^ no output
```

**Comment by lmagyar** [comment]
See #509
